src/SpinWaveTheory/SpinWaveTheory.jl

```diff
@@ -75,78 +77,79 @@ function generate_local_sun_gens(sys :: System)
-            for μ = 1:3
-                s̃_mat[:, :, μ, atom] = Hermitian(U_mat_2' * s_mat_2[μ] * U_mat_2)
-            end
-            for ν = 1:5
-                Q̃_mat[:, :, ν, atom] = Hermitian(U_mat_N' * Q_mat_N[ν] * U_mat_N)[1:2, 1:2]
-            end
-
-            if !no_single_ion
-                T̃_mat[:, :, atom] = Hermitian(U_mat_N' * sys.interactions_union[atom].aniso.matrep * U_mat_N)[1:2, 1:2]
-            end
+        for ν = 1:5
+            Q̃_mat[:, :, ν, atom] = Hermitian(U_mat' * Q_mat_N[ν] * U_mat)
         end
+        T̃_mat[:, :, atom] = Hermitian(U_mat' * sys.interactions_union[atom].aniso.matrep * U_mat)
     end
 
     return s̃_mat, T̃_mat, Q̃_mat
 end
 
-function SpinWaveTheory(sys::System{N}, energy_ϵ::Float64=1e-8, energy_tol::Float64=1e-6) where N
-    (N == 0) && error("`SpinWaveTheory` requires an SU(N) `System`. Dipole mode is not currently supported.") 
-
+function SpinWaveTheory(sys::System{N}; energy_ϵ::Float64=1e-8, energy_tol::Float64=1e-6) where N
     # Reshape into single unit cell
     cellsize_mag = cell_dimensions(sys) * diagm(collect(sys.latsize))
     sys = reshape_geometry_aux(sys, (1,1,1), cellsize_mag)
 
-    s̃_mat, T̃_mat, Q̃_mat = generate_local_sun_gens(sys)
+    # Computes the Stevens operator in the local reference frame and the SO(3) rotation matrix from global to local frame
+    # (:dipole mode only)
+    c′_coef = Vector{StevensExpansion}()
+    R_mat   = Vector{Mat3}()
+    if sys.mode == :SUN
+        s̃_mat, T̃_mat, Q̃_mat = generate_local_sun_gens(sys)
+    elseif sys.mode == :dipole
+        Nₘ, Ns = length(sys.dipoles), sys.Ns[1] # number of magnetic atoms and dimension of Hilbert 
+        s_mat_N = spin_matrices(Ns)
+        Nₘ = length(sys.dipoles)
+        s̃_mat = Array{ComplexF64, 4}(undef, 2, 2, 3, Nₘ)
+        T̃_mat = zeros(ComplexF64, 0, 0, 0)
+        Q̃_mat = zeros(ComplexF64, 0, 0, 0, 0)
+        Nₘ = length(sys.dipoles)
+        R  = zeros(Float64, 3, 3)
+        for atom = 1:Nₘ
+            θ, ϕ = dipole_to_angles(sys.dipoles[1, 1, 1, atom])
+            # U_mat = exp(-1im*ϕ*s_mat_N[3]) * exp(-1im*θ*s_mat_N[2])
+            # SO(3) rotation that aligns the quantization axis. Important: since we will project out bosons that correspond to multipolar fluctuations,
+            # therefore we use the explicit matrix to get rid of any ambiguity
+            # Note that R * (0, 0, 1) = normalize(sys.dipoles[1,1,1,atom]))
+            R[:] = [-sin(ϕ) -cos(ϕ)*cos(θ) cos(ϕ)*sin(θ);
+                     cos(ϕ) -sin(ϕ)*cos(θ) sin(ϕ)*sin(θ);
+                     0.0     sin(θ)        cos(θ)]
+            (; c2, c4, c6) = sys.interactions_union[atom].aniso.stvexp
+            SR  = Mat3(R)
+            # In Cristian's note, S̃ = R S, so here we should pass SR'
+            push!(R_mat, SR')
+            for μ = 1:3
+                s̃_mat[:, :, μ, atom] = Hermitian(rotate_operator(s_mat_N[μ], SR))[1:2, 1:2]
```

Comment:
Why is the indexing at the end necessary? (Isn't the result 2x2?)

Reply:
This is connected to the next question you are asking. Previously, I used the 2x2 matrix for the `:dipole` mode. Therefore, I must include the `M` scaling factor. Now I am using the (2S+1)x(2S+1) matrix and keeping only the first two columns/rows. In this way, the `M` factor is included. The revision looks nicer to me.

Reply:
Gotcha, I see. Thanks for explaining.